src/components/Modals/CreateSingleItemModal/CreateSingleItemModal.tsx

```diff
@@ -181,20 +181,36 @@ export default class CreateSingleItemModal extends React.PureComponent<Props, St
   }
 
   sortContentZipBothBodyShape = (bodyShape: BodyShapeType, contents: Record<string, Blob>): SortedContent => {
-    const male: Record<string, Blob> = {},
-      female: Record<string, Blob> = {}
-    for (const key in contents) {
+    let male: Record<string, Blob> = {}
+    let female: Record<string, Blob> = {}
+    let all: Record<string, Blob> = {}
+
+    for (const [key, value] of Object.entries(contents)) {
       if (key.startsWith('male/') && (bodyShape === BodyShapeType.BOTH || bodyShape === BodyShapeType.MALE)) {
-        male[key] = contents[key]
+        male[key] = value
       } else if (key.startsWith('female/') && (bodyShape === BodyShapeType.BOTH || bodyShape === BodyShapeType.FEMALE)) {
-        female[key] = contents[key]
+        female[key] = value
+      } else {
+        all[key] = value
       }
     }
-    const all = {
+
+    male = {
+      ...male,
+      ...(bodyShape === BodyShapeType.BOTH || bodyShape === BodyShapeType.MALE ? this.prefixContents(BodyShapeType.MALE, all) : {})
+    }
+
+    female = {
+      ...female,
+      ...(bodyShape === BodyShapeType.BOTH || bodyShape === BodyShapeType.FEMALE ? this.prefixContents(BodyShapeType.FEMALE, all) : {})
+    }
+
+    all = {
```

Comment:
wont this now have repeated things? with and without the prefix?

Reply:
it will have only the prefixed props.. I'll rename the var because it's confusing 😅